### PR TITLE
Add opt support for naming module prints and to diff them in dndxc.

### PR DESCRIPTION
**Pull request by marcelolr**

### CODE REVIEW

File: lib/HLSL/DxcOptimizer.cpp [hunk 1]
```diff
@@ -47,6 +47,13 @@ using namespace hlsl;
 inline static bool wcseq(LPCWSTR a, LPCWSTR b) {
   return 0 == wcscmp(a, b);
 }
+inline static bool wcsstarts(LPCWSTR value, LPCWSTR prefix) {
```
Comment (dmpots) [inline lib/HLSL/DxcOptimizer.cpp]: `startswith` would be a more common name

File: lib/HLSL/DxcOptimizer.cpp [hunk 2]
```diff
@@ -47,6 +47,13 @@ using namespace hlsl;
...
+  while (*value && *prefix && *value == *prefix) {
```
Comment (dmpots) [inline lib/HLSL/DxcOptimizer.cpp]: What if value or prefix is `nullptr` to begin with?

Comment (marcelolr) [inline lib/HLSL/DxcOptimizer.cpp]: Kaboom. Much like other str* functions. I could add an assert too, but it'll AV on contact so it should be straightforward to blame caller from that.

File: lib/HLSL/DxcOptimizer.cpp (hunk 1 above)
Comment (marcelolr) [inline lib/HLSL/DxcOptimizer.cpp]: Ack.

**Pull request merged**